File: CS/App1/RoleToFilterStringConverter.cs
using DevExpress.Data.Filtering;
using System;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Markup;

namespace App1
{

    public class RoleToFilterStringConverter : MarkupExtension, IValueConverter
    {
        protected override object ProvideValue()
        {
            return this;
        }

        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return CriteriaOperator.Parse("[Role] == ?", value);
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}

File: VB/App2/RoleToFilterStringConverter.vb
Imports DevExpress.Data.Filtering
Imports Windows.UI.Xaml.Markup

Public Class RoleToFilterStringConverter
    Inherits MarkupExtension
    Implements IValueConverter

    Protected Overrides Function ProvideValue() As Object
        Return Me
    End Function

    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal language As String) As Object Implements IValueConverter.Convert
        Return CriteriaOperator.Parse("[Role] == ?", value)
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

